website/content/en/preview/tasks/amitasks.md

```diff
@@ -0,0 +1,157 @@
+* [**Drift**]({{< relref "../concepts/disruption/#drift" >}}): Nodes are set for deletion when they drift from the desired state of the machines and new nodes are brought up to replace them.
+* [**Interruption**]({{< relref "../concepts/disruption/#interruption" >}}): Nodes are sometimes involuntarily disrupted by things like Spot interruption, health changes, and instance events, requiring new nodes to be deployed.
+
+See [**Automated Methods**]({{< relref "../concepts/disruption/#automated-methods" >}}) for details on how Karpenter uses these automated actions to replace nodes.
+
+With these types of automated updates in place, there is some risk of a new AMI being brought up that introduces some incompatibilities or bugs that cause your workloads to be degraded or fail altogether.
+The tasks described below tell you how to take more control over the ways in which Karpenter handles AMI assignments to nodes.
+
+{{% alert title="Important" color="warning" %}}
+If you are new to Karpenter, you should know that the behavior described here is different than you get with Managed Node Groups (MNG). MNG will always use the assigned AMI when it creates a new node and will never automatically upgrade to a new AMI when a new node is required. See [Updating a Managed Node Group](https://docs.aws.amazon.com/eks/latest/userguide/update-managed-node-group.html) to see how you would manually update MNG to use new AMIs.
+{{% /alert %}}
+
+## Tasks
+
+The following tasks let you have an on impact Karpenter’s behavior as it relates to how nodes are created and AMIs are consumed.
+
+### Task 1: Lock down which AMIs are selected
+
+Instead of letting Karpenter always run the latest AMI, you can change Karpenter’s default behavior.
+When you configure the [**EC2NodeClass**]({{< relref "../concepts/nodeclasses" >}}), you can override the `AMIFamily` field to chose a specific AMI and prevent a new and potentially untested AMI from replacing existing nodes when those nodes are terminated.
+
+Using the `amiSelectorTerms` field in an `EC2NodeClass`, you can set a specific AMI for Karpenter to use, based on AMI name or id (only one is required).
+Here is an example:
+
+```bash
+amiSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: "${CLUSTER_NAME}"
+      environment: prod
+  - name: al2023-ami-2023.3.20240219.0-kernel-6.1-x86_64
+  - id: ami-052c9ea013e6e3567
+```
+
+See the [**spec.amiSelectorTerms**]({{< relref "../concepts/nodeclasses/#specamiselectorterms" >}}) section of the NodeClasses page for details. 
+Keep in mind, that this could prevent you from getting critical security patches when new AMIs are available, but it does give you control over exactly which AMI is running.
+
+### Task 2: Prevent Karpenter from disrupting nodes
```

Comment:
I'm wondering if this is really a recommendation we want to make, at least in the context of managing AMIs. My main concern is that this really only helps in static clusters since the latest AMI would still be selected for new capacity launched. This could be eliminated by following Task 1 as well but I'm not sure what additional value this gives, in the context of managing AMIs, over implementing Task 1.

Reply:
Good thought. Let's bring this up for discussion when we review the doc later today.